File: backend_test/src/main/java/com/backend/passthemon/service/DemandService.java
package com.backend.passthemon.service;

import com.backend.passthemon.dto.DemandInfoDto;
import com.backend.passthemon.entity.Demand;

import java.util.List;

public interface DemandService {
    Integer addDemand(Demand demand);
    List<Demand> listAllDemandsByPage(Integer fetchPage, Integer category, Integer attrition);
    List<Demand> listOnesDemandsByPage(Integer fetchPage, Integer userId);
    Demand getDemandById(Integer demandId);
    void editDemand(DemandInfoDto demandInfoDto);
    void removeDemand(Integer demandId);
    void changeState(Integer demandId, Integer state);
}

File: frontend/src/service/DemandService.js
import {url} from '../config/url';
import {GET, POST_V2} from '../utils/Axios';

export const addDemand = (data, callback) => {
  const request = `${url}/demand/addDemand`;
  POST_V2(request, data, callback);
};

export const listAllDemandsByPage = (data, callback) => {
  const request = `${url}/demand/listAllDemandsByPage`;
  GET(request, data, callback);
};

export const listOnesDemandsByPage = (data, callback) => {
  const request = `${url}/demand/listOnesDemandsByPage`;
  GET(request, data, callback);
};

export const getDemandById = (data, callback) => {
  const request = `${url}/demand/getDemandById`;
  GET(request, data, callback);
};
export const editDemand = (data, callback) => {
  const request = `${url}/demand/editDemand`;
  POST_V2(request, data, callback);
};
export const removeDemand = (data, callback) => {
  const request = `${url}/demand/removeDemand`;
  GET(request,data,callback);
}
